### PR TITLE
Handle truncated results for iam_user enumeration

### PR DESCRIPTION
**Pull request by zleswomp**
Without this change, truncated results from the AWS API are not handled
correctly, leaving some user resources undiscovered.  This results in an
incomplete picture of what already exists.  Here we extend the iam_user
provider to cleanly handle truncated results and continue enumerating.

### CODE REVIEW

File: lib/puppet/provider/iam_user/v2.rb [hunk 1]
```diff
@@ -5,9 +5,29 @@
...
+    user_results = iam_client.list_users()
+    users = user_results.users
+
+    truncated = user_results.is_truncated
+    marker = user_results.marker
+
+    while truncated and marker
+      Puppet.debug('iam_user results truncated, proceeding with discovery')
+      response = iam_client.list_users({marker: marker})
+      response.users.each {|u|
```
Comment (bastelfreak) [inline lib/puppet/provider/iam_user/v2.rb]: ruby styleguide recommends using a oneliner here AFAIK:
```ruby
response.users.each {|u| users << u }
```

Comment (bastelfreak) [inline lib/puppet/provider/iam_user/v2.rb]: multiline do/end is also possible, but should only be used if you do multiple statements in it, like this:

```ruby
response.users.each do user
  users << user
  puts user
end
```

Comment (zleswomp) [inline lib/puppet/provider/iam_user/v2.rb]: Updated, thank you @bastelfreak.

**Comment by zleswomp** [comment]
Thanks @hunner

**Pull request merged**